### PR TITLE
Add support for case search label multimedia

### PR DESCRIPTION
**Pull request by mkangia**
## Summary

Best to review by commit.

[Ticket](https://dimagi-dev.atlassian.net/browse/USH-901)
[Requirements](https://dimagi-dev.atlassian.net/browse/USH-901?focusedCommentId=139013)

The second and final PR after https://github.com/dimagi/commcare-hq/pull/29735 to add support for multimedia for case search labels.
(This PR can be reviewed without reading the first PR)

Summary for reviewer for main commits:
This PR just builds on same lines on how we add multimedia for Case List,
First commit,
1. inherit from `NavMenuItemMediaMixin`, add a way to get_app and add reference for `_module` in `assign_references`
2. add media references in view context, with identifier, same as that passed to the template
3. add case search labels medias to module's `all_media` just like case list media. I believe this would make them available in downloads and other areas where multimedia files are accessed for a module.

Second commit,
1. fetch paths set for different media in view and save it 

Following commits are self explanatory.

For reference,
Code changes exploration done earlier in https://github.com/dimagi/commcare-hq/pull/29714

## Feature Flag

`USH_CASE_CLAIM_UPDATES`

## Product Description

Users can now set image and audio multimedia for case search labels "Search" and "Search Again".

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### QA Plan

https://dimagi-dev.atlassian.net/browse/QA-3055

### Safety story

Tested locally. Will also go through QA.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html
Comment (mkangia) [inline corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html:123]: stopBinding used because the enclosing template with bound to a different object

https://github.com/dimagi/commcare-hq/blob/8469ce862df7561c3cf86205f6a9f49804fdf45e/corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html#L9

and these are to be bound through a different mechanism

https://github.com/dimagi/commcare-hq/blob/5ca2ffe38023b0412c906d1aac1830bf0e4fa032/corehq/apps/app_manager/static/app_manager/js/nav_menu_media.js#L4-L13

File: corehq/apps/app_manager/views/view_generic.py
Comment (mkangia) [inline corehq/apps/app_manager/views/view_generic.py:197]: taking hints from how `get_case_list_menu_item_media` used a few lines above, is defined

https://github.com/dimagi/commcare-hq/blob/c46b51ca256aa4bc0d4cdeefe21f453bb199b17f/corehq/apps/hqmedia/models.py#L831-L837

the method for menu_refs, `get_case_search_label_media` would be
```
def get_case_search_label_media(self, module, label, to_language=None):
    if not module or not module.uses_media() or not hasattr(module, 'search_config'):
        return {}
    # rest of the code
```
So i just thought of getting those conditions up here itself.
If these specific media are not added there really should not be any issue since the UI would not be showing the media fields as well.

File: corehq/apps/app_manager/views/modules.py
Comment (mkangia) [inline corehq/apps/app_manager/views/modules.py:1117]: this method is called at another place already, considering making it public (removing the `_`).

File: corehq/apps/hqmedia/models.py
Comment (mkangia) [inline corehq/apps/hqmedia/models.py:678]: accidentally stumbled on this method while searching for `hasattr(self, 'case_list')` :)

File: corehq/apps/app_manager/static/app_manager/js/details/screen_config.js
Comment (mkangia) [inline corehq/apps/app_manager/static/app_manager/js/details/screen_config.js:1112]: https://github.com/dimagi/commcare-hq/pull/29735#discussion_r634715474
makes me think this is the correct flag.

It was originally set in https://github.com/dimagi/commcare-hq/pull/11438/files

File: corehq/apps/app_manager/static/app_manager/js/details/case_claim.js
Comment (orangejenny) [inline corehq/apps/app_manager/static/app_manager/js/details/case_claim.js:206]: If someone is using a custom multimedia path, and they just change the path, will the save button get triggered? There might be a generic handler that listens for changes on all `<input>` elements, I can't remember.

File: corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html
Comment (orangejenny) [reply corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html:123]: As with most self-PR comments, this would be a good code comment (maybe without the code references).

Comment (orangejenny) [reply corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html:123]: Alternatively, what would you think of adding a param to `nav_menu_media.html` to optionally include the `stopBindings`? Our custom knockout components do something similar; they have a flag indicating whether they should apply their own bindings or not.

File: corehq/apps/app_manager/static/app_manager/js/details/case_claim.js
Comment (orangejenny) [inline corehq/apps/app_manager/static/app_manager/js/details/case_claim.js:218]: Are the `|| null` parts necessary?

File: corehq/apps/app_manager/views/modules.py
Comment (orangejenny) [reply corehq/apps/app_manager/views/modules.py:1117]: I agree with removing the `_`.

File: corehq/apps/hqmedia/models.py
Comment (orangejenny) [reply corehq/apps/hqmedia/models.py:678]: Seems like it's part of ICDS's multimedia paths upload/download. I'm tempted to say we should remove that whole feature, but it looks like USH is using it.

File: corehq/apps/app_manager/static/app_manager/js/details/screen_config.js
Comment (orangejenny) [reply corehq/apps/app_manager/static/app_manager/js/details/screen_config.js:1112]: I don't think this is right. This flag controls the saving of the entire case search model: https://github.com/dimagi/commcare-hq/blob/0c8d6d5f404beca1a92d29ea8d0f1ff4e3ee95cf/corehq/apps/app_manager/static/app_manager/js/details/screen_config.js#L996-L998

That model contains SYNC_SEARCH_CASE_CLAIM features, not just USH_CASE_CLAIM_UPDATES features.

**Review by orangejenny — COMMENTED** [review]
The comment on `containsSearchConfiguration` is blocking, the rest is minor.

File: corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html
Comment (mkangia) [reply corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html:123]: I think its better this way so that one can see in the parent template itself that a section of it is not binding along with the parent. If we move it within the nested template, that visibility would go.
Though, I will be happy to make the change to pass a param if it seems better.

File: corehq/apps/app_manager/static/app_manager/js/details/screen_config.js
Comment (mkangia) [reply corehq/apps/app_manager/static/app_manager/js/details/screen_config.js:1112]: Thanks. Will revert that.

is the understanding that any domain that would have `USH_CASE_CLAIM_UPDATES` would also have `SYNC_SEARCH_CASE_CLAIM` for sure?
Because if that is not then I am quite confused.

File: corehq/apps/app_manager/static/app_manager/js/details/case_claim.js
Comment (mkangia) [reply corehq/apps/app_manager/static/app_manager/js/details/case_claim.js:206]: Good catch. Changing just the path does not enable save. I will take a look.

File: corehq/apps/app_manager/static/app_manager/js/details/case_claim.js
Comment (mkangia) [reply corehq/apps/app_manager/static/app_manager/js/details/case_claim.js:218]: I took that from other similar implementations.
Seems like the reason is that otherwise this would be empty string in case nothing is set. So that is being modified to pass null instead. I am not sure if that makes a different later but going to assume it was done for a reason i have not found that.

File: corehq/apps/app_manager/static/app_manager/js/details/screen_config.js
Comment (mkangia) [reply corehq/apps/app_manager/static/app_manager/js/details/screen_config.js]: this was the latest commit so i simply removed it for cleaner history

File: corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html
Comment (orangejenny) [reply corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html:123]: I think it can go either way, happy to defer to your judgment.

File: corehq/apps/app_manager/static/app_manager/js/details/screen_config.js
Comment (orangejenny) [reply corehq/apps/app_manager/static/app_manager/js/details/screen_config.js]: Yeah, any domain with `USH_CASE_CLAIM_UPDATES` turned on also has to have `SYNC_SEARCH_CASE_CLAIM` turned on.

**Review by orangejenny — APPROVED** [review]

**Review by orangejenny — APPROVED** [review]

**Comment by mkangia** [comment]
Hey @orangejenny 
thanks for bringing this up to date. The first commit you made here, is same as in the [PR](https://github.com/dimagi/commcare-hq/pull/29816) you made. Should it be only here considering its only in this PR we are adding LocalizedAction?

**Comment by orangejenny** [comment]
@mkangia Practically, it doesn't matter, since the bug won't be exposed until this PR is merged, but in theory it's a bug with the old code, so I'd like to get it fixed and merged.

Could you add tests to this PR?

File: corehq/apps/app_manager/static/app_manager/js/details/case_claim.js
Comment (orangejenny) [inline corehq/apps/app_manager/static/app_manager/js/details/case_claim.js:211]: I think this should be `change` because `click` won't notice if a user toggles the checkbox with their space bar.

File: corehq/apps/app_manager/static/app_manager/js/details/case_claim.js
Comment (mkangia) [reply corehq/apps/app_manager/static/app_manager/js/details/case_claim.js:207]: resolved by https://github.com/dimagi/commcare-hq/pull/29765/commits/1541eb65b5f80484d7e6ffd9336ef72589465119

File: corehq/apps/app_manager/views/modules.py
Comment (mkangia) [reply corehq/apps/app_manager/views/modules.py]: resolved by https://github.com/dimagi/commcare-hq/pull/29765/commits/89838899561e2de0f4a306ae4392c5ed8f1944e2

File: corehq/apps/app_manager/static/app_manager/js/details/case_claim.js
Comment (mkangia) [reply corehq/apps/app_manager/static/app_manager/js/details/case_claim.js]: https://github.com/dimagi/commcare-hq/pull/29765/commits/05a49f1d95e86b554b7679c6e29fa45428e3e240

I would not have thought of anyone using spacebar + tabs to get to the checkbox. took me a few tabs to get to the checkbox. :)

**Comment by mkangia** [comment]
Hey @orangejenny 

thanks for your last review.

> Could you add tests to this PR?

Added test for the suite files to account for Action/LocalizedAction change.

Also wrote test for app strings, that revealed a bug, which I fixed in https://github.com/dimagi/commcare-hq/pull/29765/commits/d65e6b4d1c66717c1afb3ef5b15da43ccae9e850

Requesting hopefully last round of review on this. This is ready to merged now since the production migration for labels is complete.

File: corehq/apps/app_manager/tests/test_suite.py
Comment (orangejenny) [inline corehq/apps/app_manager/tests/test_suite.py:184]: "new" versions 😉

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/app_manager/tests/test_suite.py
Comment (mkangia) [reply corehq/apps/app_manager/tests/test_suite.py:184]: should that be newer? :package:

Comment (mkangia) [reply corehq/apps/app_manager/tests/test_suite.py]: updated to better documentation and also to trigger a test build :)

File: corehq/apps/app_manager/tests/data/suite/search_command_detail.xml
Comment (mkangia) [inline corehq/apps/app_manager/tests/data/suite/search_command_detail.xml:19]: @orangejenny can you please help confirm if this is because of https://github.com/dimagi/commcare-hq/pull/29784/files?

I checked out to master, and added
```
# wrap to have assign_references called
self.app = Application.wrap(self.app.to_json())
# reset to newly wrapped module
self.module = self.app.modules[0]
```
in `setUp`, that gave me two sort blocks in the diff which makes me think that wrapping the application doc brings sort setting into action, though here there was a diff of 4, which is not clear why that would be.

Comment (mkangia) [reply corehq/apps/app_manager/tests/data/suite/search_command_detail.xml:19]: found the real reason on why this was happening here, fixing it here https://github.com/dimagi/commcare-hq/pull/29866.

Its expected to have these 4 sort blocks added after wrap

**Review by orangejenny — APPROVED** [review]

**Pull request merged**
